niljs/src/utils/transaction.ts

```diff
@@ -0,0 +1,70 @@
+import type { PublicClient } from "../clients/PublicClient.js";
+import { bytesToHex } from "../encoding/fromBytes.js";
+import type { Hex } from "../types/Hex.js";
+import type { ProcessedReceipt } from "../types/IReceipt.js";
+import { waitTillCompleted } from "./receipt.js";
+
+/**
+ * Represents a transaction that can be awaited for completion
+ */
+export class TransactionResponse {
+  public readonly hash: Hex;
+  constructor(
+    private readonly tx: Uint8Array,
```

Comment:
Also this is not a tx. it's just a hash in this case. So it's better to be named txHash